src/Uno.UI.Composition/Composition/VisualCollection.cs

```diff
@@ -30,7 +25,7 @@ public void InsertAbove(Visual newChild, Visual sibling)
 		{
 			var index = _visuals.IndexOf(sibling);
 			_visuals.Insert(index, newChild);
-
+			newChild.Parent = _owner;
```

Comment:
I wonder if we should be enhancing our leak testing to validate for lingering visuals (as we're making a bi-directional link with this change).

Reply:
Our leak testing has already catch an issue with this. We could definitely enhance tests for this, but apparently it's already robust enough to catch leak on visuals.